include/mxnet/tensor_blob.h

```diff
@@ -456,7 +456,7 @@ class FieldEntry<mxnet::TShape>
     this->enforce_nonzero_ = true;
     return this->self();
   }
-  inline FieldEntry<mxnet::TShape> &set_expect_ndim(mxnet::index_t ndim) {
+  inline FieldEntry<mxnet::TShape> &set_expect_ndim(int ndim) {
```

Comment:
why int? I know dimensions can't be too high but wouldn't it make sense to have a typedef for it?

Reply:
Prefer `int` to defined type for better readability. `unsigned int` is discouraged to use as data type. See https://google.github.io/styleguide/cppguide.html#Integer_Types

Reply:
I get your point. I'm not convinced that the dimensions type should be signed, as it makes no sense. Agree that the int width should be enough. I personally would make it unsigned.

Reply:
Thanks @reminisce  and @apeforest . Let's resolve this comment then.

Reply:
Adding to this, the type is signed ok I understand that. I still think we should specify the width of the type, say int32_t